lib/instance.js

```diff
@@ -351,8 +351,8 @@ module.exports = (function() {
             value = _.isString(value) ? !!parseInt(value, 10) : !!value;
           }
 
-          if (!options.raw && originalValue !== value) {
-            this._previousDataValues[key] = originalValue;
+          if (!options.raw && (options.isNewRecord || options.reset)) {
```

Comment:
Should `previous` be returning the value we think is in the database or just the last value that was assigned?